src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c

```diff
@@ -65,122 +149,999 @@ static DamageTable D_80BA82CC = {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6440.s")
+void EnRailgibud_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnRailgibud* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA64AC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6584.s")
+    this->unk_28C = 0;
+    if (this->actor.shape.shadowAlpha == 255) {
+        func_8012C28C(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6604.s")
+        gDPSetEnvColor(POLY_OPA_DISP++, 0, 0, 0, this->actor.shape.shadowAlpha);
+        gSPSegment(POLY_OPA_DISP++, 0x08, D_801AEFA0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6664.s")
+        POLY_OPA_DISP =
+            SkelAnime_DrawSV2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                              this->skelAnime.dListCount, EnRailgibud_OverrideLimbDraw, EnRailgibud_PostLimbDraw, &this->actor, POLY_OPA_DISP);
+    } else {
+        func_8012C2DC(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA66C8.s")
+        gDPSetEnvColor(POLY_XLU_DISP++, 0, 0, 0, this->actor.shape.shadowAlpha);
+        gSPSegment(POLY_XLU_DISP++, 0x08, D_801AEF88);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6800.s")
+        POLY_XLU_DISP =
+            SkelAnime_DrawSV2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                              this->skelAnime.dListCount, EnRailgibud_OverrideLimbDraw, EnRailgibud_PostLimbDraw, &this->actor, POLY_XLU_DISP);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6974.s")
+    if (this->unk_3F6 > 0) {
+        func_800BE680(globalCtx, &this->actor, this->unk_1D8, ARRAY_COUNT(this->unk_1D8), this->unk_2A4, 0.5f,
+                      this->unk_2A0, this->unk_404);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6B30.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6B9C.s")
+void func_80BA7B6C(EnRailgibud* this, GlobalContext* globalCtx) {
+    s32 pad[2];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA6D10.s")
+    func_80BA7C78(this);
+    this->unk_3FE = 99;
+    this->actor.flags |= 0x100000;
+    this->actor.flags |= 0x10;
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 28.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060053E8, &D_0600ABE0, this->jointTable, this->morphTable, 26);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    if (gSaveContext.entranceIndex != 0x2090) {
```

Comment:
```suggestion
    if (gSaveContext.entranceIndex != 0x2090) { //  Cutscene: Music Box House Opens
``` Minor suggestion for context